File: LinkedList/typescript/insert_at_head.ts
class Node<T> {
    value: T;
    next: Node<T> | null;

    constructor(value:T){
        this.value = value;
        this.next = null
    }
}

function insert_at_head(head_container:{head:Node<number>|null},value:number){
    const newNode = new Node(value)
    let temp:Node<number> | null = head_container["head"]

    if (temp == null){
        head_container["head"] = newNode
        return 
    }
    newNode.next = temp
    head_container["head"] = newNode
}

function print_node(head_container:{head:Node<number> | null}){
    let temp = head_container["head"]
    while (temp != null){
        process.stdout.write(temp.value + " ")
        temp = temp.next
    }
    console.log()
}

let head_container = {head:null}
insert_at_head(head_container,1)
insert_at_head(head_container,2)
insert_at_head(head_container,3)
insert_at_head(head_container,4)
insert_at_head(head_container,5)
print_node(head_container)

File: LinkedList/python/insert_at_head.py
class Node :
    value = None
    next = None
    
    def __init__(self,value):
        self.value = value
        self.next = None


def insert_at_head(head_container,value):
    newNode = Node(value)
    temp = head_container["head"]
    if temp is None:
        head_container["head"] = newNode
        return 

    newNode.next = temp
    head_container["head"] = newNode

def print_node(head_container):
    temp = head_container["head"]

    while temp!=None:
        print(temp.value,end=" ")
        temp=temp.next

head_container = {"head":None}
insert_at_head(head_container,1)
insert_at_head(head_container,2)
insert_at_head(head_container,3)
print_node(head_container)
